lib/models/message.ts

```diff
@@ -281,4 +293,52 @@ export class MessageModel extends DocumentDbModel<
       query: "SELECT * FROM messages m WHERE (m.fiscalCode = @fiscalCode)"
     });
   }
+
+  public async attachStoredContent(
+    blobService: BlobService,
+    messageId: string,
+    partitionKey: string,
+    message: IMessageWithContent
+  ): Promise<Either<Error, Option<DocumentDb.AttachmentMeta>>> {
+    // this is the attachment id __and__ the filename
+    const blobId = this.getMessageAttachmentName(messageId);
+
+    // store media (attachment) with message content in blob storage
+    const errorOrMessageContent = await upsertBlobFromText(
+      blobService,
+      MESSAGE_BLOB_STORAGE_CONTAINER_NAME,
+      blobId,
+      JSON.stringify({ id: messageId, ...message })
+    );
+
+    if (errorOrMessageContent.isLeft) {
+      return left(errorOrMessageContent.left);
+    }
+
+    // attach the created media to the message identified by messageId and partitionKey
+    const errorOrAttachmentMeta = await this.attach(messageId, partitionKey, {
+      contentType: "application/json",
+      id: blobId,
+      media: getBlobUrl(
+        blobService,
+        MESSAGE_BLOB_STORAGE_CONTAINER_NAME,
+        blobId
+      )
+    });
+
+    if (errorOrAttachmentMeta.isLeft) {
+      return left(
+        new Error(
```

Comment:
If you don't want to lose the `QueryError` (by converting it to an `Error`), you could change the return type of this function to:

`Promise<Either<Error | QueryError, Option<DocumentDb.AttachmentMeta>>>`

or

`Promise<Either<Either<Error, QueryError>, Option<DocumentDb.AttachmentMeta>>>`

Reply:
see https://github.com/teamdigitale/digital-citizenship-functions/pull/59#discussion_r141296878